File: java/04_016.java
public interface Coffee {
    double getCost();
}

public class SimpleCoffee implements Coffee {
    public double getCost() { return 1.0; }
}

public class MilkDecorator implements Coffee {
    private Coffee coffee;
    public double getCost() {
        return coffee.getCost() + 0.5;
    }
}

File: python/04_016.py
from abc import ABC, abstractmethod

class Coffee(ABC):
    @abstractmethod
    def get_cost(self):
        pass

class SimpleCoffee(Coffee):
    def get_cost(self):
        return 1.0

class MilkDecorator(Coffee):
    def __init__(self, coffee):
        self.coffee = coffee
    
    def get_cost(self):
        return self.coffee.get_cost() + 0.5